mentat/code_context.py

```diff
@@ -180,14 +168,20 @@ async def _get_code_message(
 
         # Setup code message metadata
         code_message = list[str]()
-        self.diff_context.clear_cache()
-        if self.diff_context.files:
-            code_message += [
-                "Diff References:",
-                f' "-" = {self.diff_context.name}',
-                ' "+" = Active Changes',
-                "",
-            ]
+        if self.diff_context:
+            self.diff_context.clear_cache()
+            if self.diff_context.files:
+                code_message += [
+                    "Diff References:",
+                    f' "-" = {self.diff_context.name}',
+                    ' "+" = Active Changes',
+                    "",
+                ]
+
+                if len(self.include_files) == 0:
+                    for file in self.diff_context.files:
+                        self.include(file)
```

Comment:
This logic used to be in `CodeContext.set_paths`. Because `set_paths` is was removed, I put it here whenever a new code message needs to be generated.

Reply:
We only want to add the diff files if the user actually did --diff or --pr-diff;
```suggestion
                if len(self.include_files) == 0 and (self.diff or self.pr_diff):
                    for file in self.diff_context.files:
                        self.include(file)
```